llvm/lib/Transforms/Instrumentation/PGOInstrumentation.cpp

```
@@ -2086,10 +2086,12 @@ static void verifyFuncBFI(PGOUseFunc &Func, LoopInfo &LI,
 
   unsigned BBNum = 0, BBMisMatchNum = 0, NonZeroBBNum = 0;
   for (auto &BBI : F) {
-    uint64_t CountValue = 0;
-    uint64_t BFICountValue = 0;
+    PGOUseBBInfo *BBInfo = Func.findBBInfo(&BBI);
+    if (!BBInfo)
+      continue;
```

Should we emit a warning in this case? Why is it ok to silently ignore here?

---

In an FDO-optimized build, hash mismatch errors are detected and logged (https://github.com/llvm/llvm-project/blob/744a469bbadb282c58e39b2c76008352f42e7e1a/llvm/lib/Transforms/Instrumentation/PGOInstrumentation.cpp#L1431-L1438).

Is it feasible to use similar mechanism to log the error here?

---

only unreachable basic blocks don't have a `BBInfo`, those shouldn't matter

---

In that case, is it possible to assert that BB is unreachable?

---

I'm guessing DominatorTree wasn't (yet) computed in -O0 and he'd rather not compute it?